api/preprint_providers/serializers.py

```diff
@@ -143,3 +147,69 @@ def update(self, instance, validated_data):
         instance.reviews_comments_anonymous = validated_data['reviews_comments_anonymous']
         instance.save()
         return instance
+
+class ModeratorSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'full_name',
+        'id',
+        'permission_group'
+    ])
+
+    id = IDField(source='_id', required=True)
+    full_name = ser.CharField(source='fullname', required=False, label='Full name', help_text='Display name used in the general user interface', max_length=186)
+    permission_group = ser.CharField(required=True)
+
+    class Meta:
+        type_ = 'moderators'
+
+    def get_absolute_url(self, obj):
+        return absolute_reverse('moderators:provider-moderator-detail', kwargs={
+            'provider_id': self.context['request'].parser_context['kwargs']['version'],
+            'moderator_id': obj._id,
+            'version': self.context['request'].parser_context['kwargs']['version']})
+
+    def create(self, validated_data):
+        user_id = validated_data.pop('_id', '')
+        if len(user_id) != 5:
+            validate_email(user_id)  # Raises ValidationError
+            full_name = validated_data.pop('fullname', '')
+            if not full_name:
+                raise ValidationError('`full_name` is required when adding a moderator via email')
+            try:
+                email = Email.objects.get(address=user_id.lower())
+            except Email.DoesNotExist:
+                user = OSFUser.create_unregistered(full_name, email=user_id)
+                user.save()
+                # TODO: send email when language exists
+            else:
+                user = email.user
+        else:
+            user = OSFUser.load(user_id)
+        if not user:
```

Comment:
Pretty sure this will only catch bad guids? Might encapsulate it with guid stuff.

Reply:
It also catches emails that are already in an `Email` object but point to no user. There are currently 0 of these, and I'm not sure that they can be created (it was written in such a way that they wouldn't be, but there could be some bug in an edge case).

Reply:
👍 makes sense